docs/markdown/snippets/compiler_argument_type.md

```diff
@@ -0,0 +1,24 @@
+## new compiler method `get_argument_type`
+
+The compiler object now has `get_argument_type` method, which returns a string
+value of `gcc`, `msvc`, or `other`. This can be used to determine if a compiler
+takes gcc style arguments `-Wfoo`, msvc style args `/w1234` or some other kind
+of arguments.
+
+```meson
+cc = meson.get_compiler('c')
+
+if cc.get_argument_type() == 'msvc'
+  if cc.has_argument('/w1235')
+    add_project_arguments('/w1235', language : ['c'])
+  endif
+elif cc.get_argument_type() == 'gcc'
+  if cc.has_argument('-Wfoo')
+    add_project_arguments('-Wfoo', language : ['c'])
+  endif
+else
+  if cc.has_argument('--error-on-foo')
+    add_project_arguments('--error-on-foo', language : ['c'])
```

Comment:
I'm not sure what kind of compiler this 'else' case is meant to apply to.

I'm not sure this example should do anything but either ignore, or raise an error in this case, since those are the only things it's safe to do.

Reply:
I'm just saying this example could set a better example.  What compiler supports '--error-on-foo'? Should you check get_id() for that, rather than assuming that *all* compilers of an unknown argument type support it?